src/v/features/enterprise_feature_messages.h

```diff
@@ -0,0 +1,119 @@
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "ssx/sformat.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+
+#include <string>
+#include <string_view>
+
+namespace features::enterprise_error_message {
+
+constexpr std::string_view required
+  = "A Redpanda Enterprise Edition license is required";
+
+constexpr std::string_view request_or_trial
+  = "To request an Enterprise license, please visit "
+    "https://redpanda.com/upgrade. To try Redpanda Enterprise for 30 days, "
+    "visit https://redpanda.com/try-enterprise.";
+
+constexpr std::string_view get_started
+  = "For more information, see "
+    "<https://docs.redpanda.com/current/get-started/licenses>.";
+
+inline ss::sstring license_nag(const auto& features) {
+    return ssx::sformat(
+      fmt::runtime(
+        "{} to use the currently enabled feature(s): ({}). Please provide a "
+        "valid license key via rpk using 'rpk cluster license set <key>', or "
+        "via Redpanda Console. {} {}}"),
+      required,
+      fmt::join(features, ", "),
+      request_or_trial,
+      get_started);
+}
+
+inline constexpr std::string upgrade_failure(const auto& features) {
+    return fmt::format(
+      "{} to use the currently enabled feature(s): ({}). To apply your "
+      "license, downgrade this broker to the pre-upgrade version and provide a "
+      "valid license key via rpk using 'rpk cluster license set <key>', or via "
+      "Redpanda Console. {} {}",
+      required,
+      fmt::join(features, ", "),
+      request_or_trial,
+      get_started);
+}
+
+inline ss::sstring cluster_property(std::string_view name, auto value) {
+    return ssx::sformat(
+      fmt::runtime(
+        "Rejected {}: '{}' '{}' is restricted to clusters with an Enterprise "
+        "license. {}"),
+      name,
+      value,
+      name,
+      request_or_trial);
+}
+
+inline ss::sstring partition_autobalancing_continuous() {
+    return ssx::sformat(
+      R"({} to use feature "partition_autobalancing_mode" with value "continuous".  Behavior is restricted to "node_add". {})",
+      required,
+      request_or_trial);
+}
+
+inline ss::sstring core_balancing_continuous() {
+    return ssx::sformat(
+      R"({} to use enterprise feature "core_balancing_continuous".  This property is being ignored. {})",
+      required,
+      request_or_trial);
+}
+
+inline ss::sstring default_leaders_preference() {
+    return ssx::sformat(
+      R"({} to use the enterprise feature "leadership pinning".  This feature is disabled.  The values of the cluster property <value> and the topic property "redpanda.leaders.preference" are being ignored. {})",
```

Comment:
> `<value>`

Should that contain something specific?

Reply:
Not according to the requirements.

Reply:
For this specific one we didn't include the values in the previous version. The issue is that to report all the potential values that might be in violation you have the enterprise property and all the topics that might have set their property to anything not-default.
Printing all the topics that are in violation could (theoretically) lead to overallocation warning.
One option was to limit the number of topic names being printed, but that required a bit of extra logic.
In the end it was decided to not print any of the values for simplicity.